subs/pantry/src/Pantry/Storage.hs

```diff
@@ -145,12 +150,20 @@ CacheUpdate
 -- table.
 Tree
     key BlobId
-    cabal BlobId
+    cabal BlobId Maybe
     cabalType FileType
     name PackageNameId
     version VersionId
     UniqueTree key
 
+HPack
+   tree TreeId
+   version VersionId
+   -- Corresponding cabal file of hpack
+   cabal BlobId
+   path FilePathId
```

Comment:
Is this the package.yaml path name, or the generated cabal file name?

Reply:
It's the generated cabal file name. For hpack based builds, `package.yaml` will anyway be present in `TreeEntry`.

Reply:
Please add a comment at least explaining that this is the cabal file path. Changing to `cabalPath` may be even better.

Reply:
I meant calling `path` instead `cabalPath` would be a clearer naming here.